### DISCUSSION TITLE
Roll call vote data for Swedish Riksdag?

### DISCUSSION THREAD

**Discussion by potmo** [swerik-project/the-swedish-parliament-corpus#29] [Q&A] [answered]
Hi,

I'm working with an art project thats been ongoing for about ten years now where I'm trying to run roll call vote data on the Swedish parliament through DW-Nominate (Poole and Rosenthal) to figure out if Olof Palme was left or right in relation to historical and current Social democrats. Although there of course is a lot of other interesting information that can be extracted from the same data I'm only looking for a yes/no answer if Palme was more left or right than todays Social democrats for this project.

Back in 2015 I contacted Keith T Poole who I think was at the University of Georgia the time and he gave me some pointers on how to setup DW-Nominate but there wasn't any raw data from the Swedish Riksdag.
I contacted the Riksdagsbibliotek to see if they had any digitised roll call vote records and did quite a lot of research into my self scanning the paper records they had. At that time there where talks about starting a project to scan the paper records by the Riksdag and Riksdagsbiblioteket which they later went through with. After a few years I got a hold of the scanned records in jpeg format.

From that I developed a software that could semi-automatically scan and parse the hole punch style roll call records that was used between 1935 to 1966 (and could probably be adapted to 1967-1983 since they are quite similar in style). I couldn't find a comprehensive list of MPs so I manually entered the MPs for each seat and figured out who was who (sometimes there are multiple Johanssons for example for the Social Democrats so I'd have to back track where they were seated from their constituency). This has resulted in a database of MPs (names, aliases, constituencies and ID:s) and votes for all roll call protocol for second chamber between 1957 to 1966 and for the first chamber 1957 to 1959. The main reason that I didn't just do all of them was that it was very tedious to create the MP database to reference from the roll call records. With the seating information it would just take a few days to go though the rest of the data.

Almost two years ago I was contacted by Jan Teorells group at Stockholm University who said that they where working on a similar project to parse the voting records and asked If I wanted to collaborate but it never got to anything. That project was supposed to end in the spring of '23 so I stopped my work since I figured I could just use their data when that was done. To prepare for that I downloaded the data that Riksdagen has and started to create a database of their MPs and votes. All of the Riksdags records does not contain MP ids so its impossible to know if is the same person or just the same name so now I have a normalized database of all the MPs from riksdagens data.

This spring I talked to Teorells group and asked them what their progress was and how they would identify the MPs and they said that they would use the Swerik MP ids and that they where delayed and did not when it would be finished.
I looked through the Swerik github and I cant find any roll call data. 

Does anyone know if Teorells group are going to publish their roll call data on Swerik or if they are just using the MP ids? Or maybe they have already published it and I just can find it?

Since the big work for me left on my own project is creating a reliable MP database (that Swerik already seems to have) I could just merge the MP database(s) I already have with the Sweriks and be done quite quickly but I've already done quite a lot of unnecessary work so if its already been done I think i'd pass on that.

**Comment by fredrik1984** [comment, marked as answer]
Hi Nisse! Sounds like an interesting project! Yes, the Swerik project has been collaborating with Jan's roll call data project. We helped him mapping MPs to their respective chairs and with other relevant MP metadata, and we have got the MP chair data in return. However, I don't think we have got the actual roll call data, not yet at least, right @BobBorges?

**Reply by MansMeg** [reply]
No, we dont have the roll call votes as of now. Jan has that now, but I think we would be happy to add it further down the line. Im not sure what database you have, but I think Jans project will use our database as you say. So if you want to use your database I guess you need to merge, but I guess our database (ie wikidata) should probably already contain everything you need?

**Comment by BobBorges** [comment]
Hi! Exactly as @MansMeg said, we don't have the role calls in our data set, only the seat numbers.

**Comment by potmo** [comment]
Okay great thats good to know. Then I should make sure i can reference Swerik MP ids from my ids that I have in my home made databases. I'll probably also download your seating info and check that agains my seating info and when Teorells publishes the roll call data I can merge that with the roll call data that I have made.

**Comment by MansMeg** [comment]
Yes. It would be great if you could share e.g. fixes you do in the chairs as a pull request.
